mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/InformationToEmail.java

```diff
@@ -0,0 +1,200 @@
+
+import org.apache.commons.io.FileUtils;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.quota.CompareWithCurrentThreshold;
+import org.apache.james.mailbox.quota.QuotaCount;
+import org.apache.james.mailbox.quota.QuotaSize;
+import org.apache.james.mailbox.quota.model.QuotaThreshold;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class InformationToEmail {
+
+    public static class Builder {
+        private Optional<QuotaThreshold> countThreshold;
+        private Optional<QuotaThreshold> sizeThreshold;
+        private Quota<QuotaSize> sizeQuota;
+        private Quota<QuotaCount> countQuota;
+
+        public Builder() {
+            countThreshold = Optional.empty();
+            sizeThreshold = Optional.empty();
+        }
+
+        public Builder sizeQuota(Quota<QuotaSize> sizeQuota) {
+            this.sizeQuota = sizeQuota;
+            return this;
+        }
+
+        public Builder countQuota(Quota<QuotaCount> countQuota) {
+            this.countQuota = countQuota;
+            return this;
+        }
+
+        public Builder countThreshold(QuotaThreshold countThreshold,
+                                      CompareWithCurrentThreshold compareWithCurrentCountThreshold) {
+            this.countThreshold = filterThresholdIfNoticeNotNeeded(countThreshold, compareWithCurrentCountThreshold);
+            return this;
+        }
+
+        public Builder sizeThreshold(QuotaThreshold sizeThreshold,
+                                     CompareWithCurrentThreshold compareWithCurrentSizeThreshold) {
+            this.sizeThreshold = filterThresholdIfNoticeNotNeeded(sizeThreshold, compareWithCurrentSizeThreshold);
+            return this;
+        }
+
+        private Optional<QuotaThreshold> filterThresholdIfNoticeNotNeeded(QuotaThreshold threshold, CompareWithCurrentThreshold compareWithCurrentThreshold) {
+            switch (compareWithCurrentThreshold) {
+                case ABOVE_CURRENT_THRESHOLD:
+                    return threshold.nonZero();
+                case BELOW_CURRENT_THRESHOLD:
+                case ABOVE_CURRENT_THRESHOLD_WITH_RECENT_CHANGES:
+                case NO_CHANGES:
+                    return Optional.empty();
+            }
+            return Optional.empty();
+        }
+
+        public Optional<InformationToEmail> build() {
+            Preconditions.checkNotNull(sizeQuota);
+            Preconditions.checkNotNull(countQuota);
+
+            if (sizeThreshold.isPresent() || countThreshold.isPresent()) {
+                return Optional.of(
+                    new InformationToEmail(countThreshold, sizeThreshold, sizeQuota, countQuota));
+            }
+            return Optional.empty();
+        }
+    }
+
+    public static class MessageBuilder {
+        public static final String PREAMBLE = "You receive this email because you recently exceeded a threshold related " +
```

Comment:
we should use a mail template with variables to replace

Reply:
that an other task I guess